src/test/scala/com/fulcrumgenomics/commons/io/IoTest.scala

```diff
@@ -189,6 +212,19 @@ class IoTest extends UnitSpec {
     an[IllegalArgumentException] should be thrownBy Io.readLinesFromResource("/path/does/not/exist.json")
   }
 
+  "Io.readLines" should "read Unix fifos" in { 
+    assume(hasMkfifo, "Canceling Posix specific test.")
+    val lines    = Seq("foo", "bar")
+    val pipeDir  = tmpdir()
+    val pipePath = pipeDir.resolve("pipe2")
+    Files.deleteIfExists(pipePath) // in case of a failed previous test that needs clean up
+    pipe(pipePath, lines)
+    val writeFuture = Future { Io.writeLines(pipePath, lines); true }
+    Io.readLines(pipePath).toList should contain theSameElementsInOrderAs lines
+    Await.result(writeFuture, Duration(1, TimeUnit.SECONDS)) shouldBe true
```

Comment:
I'd maybe bump up the wait time here to a few seconds just so we don't get sporadic test failures if the travis instance is slow or pauses.

Reply:
Bumped up up time of waiting, added test for BufferedInputStream for both symlink and real files